### PR TITLE
[AND-3314] Refactor vungle banner adapter. 

### PR DESCRIPTION
**Pull request by felixzhang0703**
https://vungle.atlassian.net/browse/AND-3314

Vungle::getNativeAd is deprecated in SDK 6.10.x, we should use banner api.

### CODE REVIEW

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java [hunk 1]
```diff
@@ -254,26 +254,7 @@ private void createBanner() {
...
-      View adView = null;
-      VungleNativeAd vungleMRECBanner = Vungle.getNativeAd(placementId, mAdConfig, playAdCallback);
-      if (vungleMRECBanner != null) {
-        adView = vungleMRECBanner.renderNativeView();
-      }
-
-      if (adView != null) {
-        Log.d(TAG, "display MREC:" + vungleMRECBanner.hashCode() + this);
-        if (vungleBannerAd != null) {
-          vungleBannerAd.setVungleMRECBanner(vungleMRECBanner);
```
Comment (alexander-gubov) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java:266]: I think VungleBannerAd should be cleaned from VungleNativeView as well (or VungleBanner can even replace VungleBannerAd)

Comment (felixzhang0703) [reply ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java:266]: OK, will dig into this part. Thanks!

Comment (felixzhang0703) [reply ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java:266]: @alexander-gubov Hi, I've checked the code and VungleBannerAd can be replaced.
Now VungleManager manages VungleBannerAdapter directly. VungleBannerAdapter will hold a VungleBanner to control VungleBanner attach, detach and destroy.

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleManager.java [hunk 2]
```diff
@@ -24,7 +24,7 @@
 
   private static VungleManager sInstance;
 
-  private ConcurrentHashMap<String, VungleBannerAd> mVungleBanners;
+  private ConcurrentHashMap<String, VungleBannerAdapter> mVungleBanners;
```
Comment (alexander-gubov) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleManager.java:27]: I know that VungleBannerAdapter is not a real adapter, but Google didn't want to have any strong reference to adapter from this map.
We need to be sure that VungleBannerAdapter don't hold any strong reference to real VungleInterstitialAdapter

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleManager.java [hunk 3]
```diff
@@ -158,8 +163,8 @@ boolean isValidPlacement(String placementId) {
    */
   private void cleanLeakedBannerAdapters() {
     for (String placementId : new HashSet<>(mVungleBanners.keySet())) {
-      VungleBannerAd bannerAd = mVungleBanners.get(placementId);
-      if (bannerAd != null && bannerAd.getAdapter() == null) {
+      VungleBannerAdapter bannerAd = mVungleBanners.get(placementId);
```
Comment (alexander-gubov) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleManager.java:166]: this seems to remove a workaround for leaking VungleInterstialAdapter(which holds a strong ref to VungleBannerAdapter and "bannerAd.getAdapter()" was a check to checks if VungleInterstialAdapter was GCetd)
please revert. If mVungleBanners contains VungleBannerAdapter, and VungleInterstialAdapter is GCted we can't figure it out.
@HuFangshuai

**Comment by alexander-gubov** [comment]
shouldn't be the target branch the same as in https://github.com/Vungle/googleads-mobile-android-mediation/pull/88 ?

**Comment by palrahul** [comment]
> shouldn't be the target branch the same as in #88 ?

No dont think so, this looks right. The changes we made for MREC etc to use Banners API will be going in SDK 6.10.1 and hence its not related to 6.9.x-release branch

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleBannerAd.java [hunk 4]
```diff
@@ -1,121 +0,0 @@
-package com.google.ads.mediation.vungle;
```
Comment (HuFangshuai) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleBannerAd.java:1]: Why we removed this class? Propose only remove VungleNativeAd related code parts.

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleManager.java (hunk 3 above)
Comment (HuFangshuai) [reply ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleManager.java:166]: Yes. Hi @felixzhang0703 here's the related PR https://github.com/googleads/googleads-mobile-android-mediation/pull/304

**Comment by felixzhang0703** [comment]
Will create a new pull request because this PR need to revert.

**Pull request closed without merging**